modules/auxiliary/gather/wp_all_in_one_migration_export.rb

```diff
@@ -0,0 +1,69 @@
+##
+# This module requires Metasploit: http://www.metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+  include Msf::HTTP::Wordpress
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(
+      info,
+      'Name'            => 'WordPress All-in-One Migration Export',
+      'Description'     => %q(Due to lack of authenticated session verification
+                              it is possible for unauthenticated users to export
+                              a complete copy of the database, all plugins, themes
+                              and uploaded files.),
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'James Golovich',                  # Disclosure
+          'Rob Carr <rob[at]rastating.com>'  # Metasploit module
+        ],
+      'References'      =>
+        [
+          ['WPVDB', '7857'],
+          ['URL', 'http://www.pritect.net/blog/all-in-one-wp-migration-2-0-4-security-vulnerability']
+        ],
+      'DisclosureDate'  => 'Mar 19 2015'
+    ))
+
+    register_options(
+      [
+        OptInt.new('MAXTIME', [ true, 'The maximum number of seconds to wait for the export to complete', 300 ])
```

Comment:
There is HTTPClientTimeout that performs this functionality as can be seen at https://github.com/rapid7/metasploit-framework/blob/c201955fdf0402f5ffe52566aed7c0a95ed45cc3/lib/msf/core/exploit/http/client.rb#L56. The advantage is you won't have to register an additional datastore option. The down side is the description won't be as good as this one. Up to you to determine which one is better. IMO we can register the HTTPClientTimeout and mentioning in the description if the user wants to wait for longer to adjust.